File: app/src/main/java/com/justaclock/viewmodels/ChronometerViewModel.java
package com.justaclock.viewmodels;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;
import com.justaclock.adapters.Task;
import java.util.ArrayList;

public class ChronometerViewModel extends AndroidViewModel {
    public MutableLiveData<ArrayList<Task>> tasks;
    public boolean                          chronometerIsRunning;
    public long                             timeWhenPause;
    public String                           lastTimeString;

    public ChronometerViewModel(@NonNull Application application) {
        super(application);
        tasks = new MutableLiveData<>();
        tasks.setValue(new ArrayList<>());
        tasks.getValue().add(new Task("Example task", "00:00:00", true));
        tasks.getValue().add(new Task("Example task", "00:00:00", true));
    }

    public void insertTask(Task task){
        tasks.getValue().add(new Task(task.getName(), task.getTime(), false));
        tasks.setValue(tasks.getValue());
    }
}

File: app/src/main/java/com/justaclock/viewmodels/ChronometerViewModel.kt
package com.justaclock.viewmodels

import android.app.Application
import android.text.Editable
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.justaclock.adapters.Task
import java.util.ArrayList

class ChronometerViewModel(application: Application) : AndroidViewModel(application) {
    var tasks: MutableLiveData<ArrayList<Task>> = MutableLiveData()
    var chronometerIsRunning: Boolean = false
    var timeWhenPause: Long = 0
    var lastTimeString: String? = null
    var base: Long? = null
    var focusMode: Boolean = false
    var beforeItWasOnPause: Boolean = false
    var taskName: MutableLiveData<String?> = MutableLiveData()

    init {
        tasks.value = ArrayList()
        tasks.value!!.add(Task("Example task", "00:00:00", true))
    }

    fun insertTask(task: Task) {
        tasks.value!!.add(Task(task.name, task.time, false))
        tasks.value = tasks.value
    }
}
